DOCS-3693: Update module and namespace renaming (#4418)
viamrobotics/docs: docs/operate/reference/naming-modules.md

To create a new namespace for your organization, click on the org's **Settings** in the top right of the navigation bar, then click the **Set a public namespace** button.
Enter a name or use the suggested name for your namespace, and then click **Set namespace**.
A namespace may only contain letters, numbers, and the dash (`-`) character.

## Update a namespace for your organization

You can change your organization's namespace on your organization settings page:

1. Navigate to your organization settings page using the dropdown in the upper right corner of the web UI.
1. Click the **Rename** button next to your current namespace.
1. Enter the new namespace name in the modal that appears.
1. Click **Set namespace** to confirm the change.

> 1. For each module your organization owns, update the module code and <file>meta.json</file> to reflect the new namespace.

Reviewer: @michaellee1019 questions:
1. If this is recommended but not required, why not?
       a. If you push updates to a module without updating namespace, does it break things?
2. Scope doc only mentions public modules. What about private modules?

When you rename a namespace, Viam reserves the old namespace for backwards compatibility and you cannot reuse it.

> Machine configurations that reference the old namespace will continue to work.

Reviewer: @michaellee1019 Here https://docs.viam.com/operate/get-started/other-hardware/manage-modules/#transfer-ownership-of-a-module we say 

> Update the model field in the configuration of any machines that use the module to use the new organization’s namespace.

But is this no longer necessary after this eng work?

Reviewer: Per scope doc comment convo, I guess it was never necessary just recommended. I've added a fix to this PR.

To create a new namespace for your organization, click on the org's **Settings** in the top right of the navigation bar, then click the **Set a public namespace** button.
Enter a name or use the suggested name for your namespace, and then click **Set namespace**.
A namespace may only contain letters, numbers, and the dash (`-`) character.

## Update a namespace for your organization

> If your organization already has a namespace and you need to change it, you can rename it through the organization settings page.

Reviewer: ```suggestion
You can change your organization's namespace on your organization settings page:
```
Okay sure, and I can eliminate the next sentence. Promptless not really showing its powers of concision on this one...

To rename your organization's namespace:

1. Navigate to your organization settings at `app.viam.com/organization/<org-id>/settings`.
1. Click the **Rename** button next to your current namespace.
1. Enter the new namespace name in the modal that appears.
1. Click **Set namespace** to confirm the change.

> Notes:
> 
> - The old namespace is permanently reserved for backwards compatibility and cannot be reused.
> - Existing machine configurations will continue to work without any changes required.
> - You need to change model names for public modules after a namespace change.
> - Organizations are limited to 5 namespaces total to prevent abuse.
> - When adding modules to new machines, the machine builder will use the new namespace.

Reviewer 2: ```suggestion
When you rename a namespace, the old namespace name cannot be reused.
References to the old namespace name will continue to work.
You can rename a namespace no more than 5 times.
```
optional: similar reasoning as above, though a bulleted list would work fine here as well (not doing that for consistency with the above section). It might be good to fact check the '5 namespaces' limit -- does that mean you can _rename_ 5 times, or that you can rename only 4 times, or that when you rename the 5th time, the oldest namespace alias will stop working? That's a really important detail!

One caveat: I don't know what this line means, but it seems awfully important:

> You need to change model names for public modules after a namespace change.

Seems to directly contradict the 'existing machine configureations will continue to work without any changes needed' line. But big if true. So someone really ought to investigate.

Reviewer: Oof looking back at the scope doc (negating the usefulness of the tool), I'm going to take the 5 namespaces limit out of here as it "won't be advertised." But fair question--Answer is here https://github.com/viamrobotics/app/pull/8224/files#r2056776563--they can't rename a 5th time.

You're right. Thanks for catching, and I'm sorry I didn't catch--this does seem contradictory (and comes directly from the scope doc). I'm asking Michael.

Reviewer: ```suggestion
1. For each module your organization owns, update the module code and <file>meta.json</file> to reflect the new namespace.

When you rename a namespace, Viam reserves the old namespace for backwards compatibility and you cannot reuse it.

Machine configurations that reference the old namespace will continue to work.
```
